src/main/java/uk/gov/hmcts/dts/fact/services/CourtService.java

```diff
@@ -126,6 +136,20 @@ public Court getCourtBySlug(final String slug) {
             .collect(toList());
     }
 
+    public List<CourtReferenceWithDistance> getNearestCourtReferencesByPostcode(final String postcode) {
+        final Optional<MapitData> optionalMapitData = mapitService.getMapitData(postcode);
+
+        if (optionalMapitData.isEmpty()) {
+            LOGGER.error("No mapit data found for provided postcode: {}", postcode);
+            return new ArrayList<>(); // Return empty so the frontend logic can be invoked
```

Comment:
In other court service methods within this class, we return emptyList() rather than new ArrayList<>(). May be we should keep it consistent